### ISSUE TITLE
`Detector` から `timer` パラメータを除去

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#47]
## Summary

`Detector` クラスから `timer: InferenceTimer` パラメータと関連ロジックを除去し, シンプルな検出 API に特化させる.

## Background

Issue #42 Step 5 で `infer` コマンドが `DetectionPipeline` + `PhasedTimer` ベースに移行した.
`DetectionPipeline` が preprocess / inference / postprocess の全フェーズを画像ごとに計測するため, `Detector` が独自に `InferenceTimer` を保持して推論フェーズだけを計測する設計は役割が重複している.

なお `InferenceTimer` クラス自体は `PhasedTimer` の内部部品として引き続き使用されており, 削除対象ではない.
あくまで `Detector` が `timer` パラメータで `InferenceTimer` を直接受け取る設計が不要になった.

## Current Implementation

`Detector.__init__` が `timer: InferenceTimer | None` を受け取り, `detect()` 内で推論フェーズのみを `with self._timer.measure():` で計測している.

```python
# pochidetection/scripts/rtdetr/inference/detector.py
class Detector:
    def __init__(
        self,
        ...
        timer: InferenceTimer | None = None,  # 削除対象
        ...
    ) -> None:

    def detect(self, image: Image.Image) -> list[Detection]:
        ...
        if self._timer is not None:
            with self._timer.measure():  # 削除対象
                pred_logits, pred_boxes = self._backend.infer(inputs)
                self._backend.synchronize()
        ...

    @property
    def timer(self) -> InferenceTimer | None:  # 削除対象
        return self._timer
```

## Proposed Changes

1. `Detector.__init__` から `timer` パラメータを削除
2. `detect()` 内の `self._timer.measure()` 分岐を削除し, 常に直接 `backend.infer()` を呼ぶ
3. `timer` プロパティを削除
4. `tests/test_scripts/test_detector.py` から `timer` 関連テストを削除・更新

## Acceptance Criteria

-  `Detector` から `timer` パラメータ, `_timer` 属性, `timer` プロパティが除去されている
-  `detect()` 内の計測分岐が除去され, 常に `backend.infer()` + `backend.synchronize()` を呼ぶ
-  `uv run pytest` が全件パスする
-  `uv run pre-commit run --all-files` がパスする

## Labels

`refactor`

## Branch

`refactor/remove-detector-timer`
